### PR TITLE
`InitSuites` changes to order making `BUILD_TLS_AES_256_GCM_SHA384` be prioritized over `BUILD_TLS_AES_128_GCM_SHA256`

### PR DESCRIPTION
**Pull request by aidangarske**
# Description 

`internal.c` - `InitSuites` changes to order making `BUILD_TLS_AES_256_GCM_SHA384` be prioritized over `BUILD_TLS_AES_128_GCM_SHA256` to match TLS 1.2 order. 
-fix for test `quic.c`, `api.c`, and `test.h` hard coded size assumption. 
-fix `ssl.c` TRUE/FALSE to 1/0. 

# Testing

```
./configure --enable-all --enable-debug --disable-shared 
make
make check
```

Requires: https://github.com/wolfSSL/osp/pull/213

### CODE REVIEW

**Comment by dgarske** [comment]
Retest this please

File: tests/api.c [hunk 1]
```diff
@@ -6495,15 +6495,10 @@ static int test_wolfSSL_EVP_CIPHER_CTX(void)
...
-    #ifdef WC_SHA512_DIGEST_SIZE
-        #define MD_MAX_SIZE WC_SHA512_DIGEST_SIZE
-    #else
-        #define MD_MAX_SIZE WC_SHA256_DIGEST_SIZE
-    #endif
-    byte server_side_msg1[MD_MAX_SIZE] = {0};/* msg sent by server */
-    byte server_side_msg2[MD_MAX_SIZE] = {0};/* msg received from client */
-    byte client_side_msg1[MD_MAX_SIZE] = {0};/* msg sent by client */
-    byte client_side_msg2[MD_MAX_SIZE] = {0};/* msg received from server */
+    byte server_side_msg1[WC_MAX_DIGEST_SIZE] = {0};/* msg sent by server */
```
Comment (SparkiDev) [inline tests/api.c:6498]: Initialization like that isn't always reliable.
Make sure an XMEMSET is done.

**Review by SparkiDev — CHANGES_REQUESTED** [review]

**Review by SparkiDev — APPROVED** [review]

**Review by dgarske — APPROVED** [review]
Please fix the Python patch in OSP to support either 32 or 48 for now. Once that is pushed through we can merge this one

**Comment by dgarske** [comment]
Retest this please. Python test should pass now.

**Comment by dgarske** [comment]
Retest this please. Generic config “ FAIL scripts/openssl.test ”

**Review by JacobBarthelmeh — APPROVED** [review]

**Comment by hholzgra** [comment]
See also: https://jira.mariadb.org/browse/MDEV-35579

**Pull request merged**